library/alloc/src/string.rs

```diff
@@ -139,6 +139,12 @@ use crate::vec::Vec;
 /// [`bytes`]: str::bytes
 /// [`chars`]: str::chars
 ///
+/// You can, however, index into a `String` in constant-time with anything that
```

Comment:
Here, again, maybe "index" is a dangerous word ... not something I had considered until reviewing my own work up here :sweat_smile: